flink-java/src/main/java/org/apache/flink/api/java/utils/RequiredParameter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Facility to manage required parameters in user defined functions.
+ */
+public class RequiredParameter {
+
+	private static final String HELP_TEXT_PARAM_DELIMITER = "\t";
+	private static final String HELP_TEXT_LINE_DELIMITER = "\n";
+
+	private HashMap<String, Option> data;
+
+	public RequiredParameter() {
+		this.data = new HashMap<>();
+	}
+
+	public void add(Option option) {
+		this.data.put(option.getName(), option);
+	}
+
+	/**
+	 * Check if all parameters defined as required have been supplied.
+	 *
+	 * @param parameterTool - parameters supplied by user.
+	 */
+	public void check(ParameterTool parameterTool) throws RequiredParameterException {
+		for (Option o : data.values()) {
+			// if the parameter is not present or its value is undefined, throw a RuntimeException.
+			if (!parameterTool.data.containsKey(o.getName()) || keyIsUndefined(o.getName(), parameterTool.data)) {
```

Comment:
The RequiredParameter tool should check for each Option that:
- the ParameterTool contains a value for either the full or short key (if short key is defined)
- the ParameterTool does not contain values for both full and short key (if short key is defined)
- if the value is set on the short key, it should be copied to the full key

The checkAndPopulate method should only add the default value to the full key

That way, the value can be retrieved using the full key.

What do you think?